nilearn/interfaces/fmriprep/load_confounds_utils.py

```diff
@@ -90,46 +90,74 @@ def _add_suffix(params, model):
 
 def _get_file_name(nii_file):
     """Construct the raw confound file name from processed functional data."""
+    import itertools
+
     if isinstance(nii_file, list):  # catch gifti
         nii_file = nii_file[0]
+
+    base_dir = os.path.dirname(nii_file)
+
+    # Build a list of potential confounds filenames using all combinations of
+    # the entities in the image file.
     entities = parse_bids_filename(nii_file)
-    subject_label = f"sub-{entities['sub']}"
-    if "ses" in entities:
-        subject_label = f"{subject_label}_ses-{entities['ses']}"
-    specifiers = f"task-{entities['task']}"
-    if "run" in entities:
-        specifiers = f"{specifiers}_run-{entities['run']}"
-    img_filename = nii_file.split(os.sep)[-1]
+    file_fields = entities["file_fields"]
+    entities = {k: v for k, v in entities.items() if k in file_fields}
```

Comment:
confound files do not include the space entity (in fmriprep at least), so maybe we do not need to include it here.

don't think it makes a big difference in terms of performance to leave it in

also make the whole thing more general to keep it in

what's your take on this? Also you don't have to have an opinion... 😝

Reply:
I don't have a strong opinion on this. I think the fewer constraints we put on this code, the less likely it will break, but it will also be marginally slower. Maybe we can keep as-is in this PR and add restrictions if users complain?

Reply:
good with me. :+1: